plugins/catalog-backend-module-incremental-ingestion/src/database/IncrementalIngestionDatabaseManager.ts

```diff
@@ -546,12 +560,28 @@ export class IncrementalIngestionDatabaseManager {
    * @param entities - DeferredEntity[]
    */
   async createMarkEntities(markId: string, entities: DeferredEntity[]) {
+    const refs = entities.map(e => stringifyEntityRef(e.entity));
+
     await this.client.transaction(async tx => {
+      const existingRefs = new Set(
+        ...(
+          await tx<{ ref: string }>('ingestion_mark_entities')
+            .select('ref')
+            .whereIn('ref', refs)
+        ).map(e => e.ref),
+      );
+
+      const newRefs = refs.filter(e => !existingRefs.has(e));
+
+      await tx('ingestion_mark_entities')
+        .update('ingestion_mark_id', markId)
+        .whereIn('ref', Array.from(existingRefs));
```

Comment:
Oh right, we want to use it as an array too. Alright, better then to have two variables - make it as an array first, and then make a set of that array in a second variable, like `existingRefsSet` or whatnot. That way we don't have to deconstruct the set again.

Reply:
Done!